pkg/controller/upload-controller.go

```diff
@@ -514,11 +574,18 @@ func UploadPossibleForPVC(pvc *v1.PersistentVolumeClaim) error {
 
 // GetUploadServerURL returns the url the proxy should post to for a particular pvc
 func GetUploadServerURL(namespace, pvc, uploadPath string) string {
-	return fmt.Sprintf("https://%s.%s.svc%s", getUploadResourceName(pvc), namespace, uploadPath)
+	serviceName := createUploadServiceNameFromPvcName(pvc)
+	return fmt.Sprintf("https://%s.%s.svc%s", serviceName, namespace, uploadPath)
+}
+
+// createUploadServiceName returns the name given to upload service shortened if needed
+func createUploadServiceNameFromPvcName(pvc string) string {
```

Comment:
This is something that may actually be nice to store in the PVC as well.  Then the upload proxy can use that value.  Understand if you don't want to tackle that right now.

Reply:
I would create another task for this, small and focused ;)